.github/workflows/benchmark.yml

```
@@ -0,0 +1,115 @@
+name: Benchmark
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Base branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.base_ref }}
+          submodules: 'recursive'
+          fetch-depth: 0
+        
+      - name: Install build tools and dependencies
+        run: sudo apt-get update && sudo apt-get install cmake ninja-build clang libgl-dev libgles-dev catch2
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          key: test-ccache-${{ runner.os }}-${{ runner.arch }}
+
+      - name: Build (Base branch)
+        run: |
+            cmake -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Debug -DBUILD_JVM=OFF -DBUILD_STANDALONE=OFF -B build-debug
+            cmake --build build-debug -- tests
+
+      - name: Run benchmarks (Base branch)
+        run: |
+            cd shared/test      # for access to data/ directory
+            ../../build-debug/shared/test/tests \
+              --reporter xml > ../../base_results.xml
+            cd ../..
+            rm -rf build-debug
+
+      - name: Checkout PR branch
+        run: |
+          git fetch origin pull/${{ github.event.pull_request.number }}/head:pr-branch
+          git checkout pr-branch
+          git submodule update --init --recursive
+
+      - name: Build (PR branch)
+        run: |
+            cmake -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Debug -DBUILD_JVM=OFF -DBUILD_STANDALONE=OFF -B build-debug
+            cmake --build build-debug -- tests
```

Same settings as before in the same build directory; don't need to repeat this.
```suggestion
            cmake --fresh build-release
            cmake --build build-release -- tests
```